internal/rbd/rbd_util.go

```diff
@@ -1043,18 +1043,16 @@ func cleanupRBDImageMetadataStash(path string) error {
 }
 
 // resize the given volume to new size.
-func (rv *rbdVolume) resize(ctx context.Context, cr *util.Credentials) error {
-	mon := rv.Monitors
-	volSzMiB := fmt.Sprintf("%dM", util.RoundOffVolSize(rv.VolSize))
-
-	args := []string{"resize", rv.String(), "--size", volSzMiB, "--id", cr.ID, "-m", mon, "--keyfile=" + cr.KeyFile}
-	_, stderr, err := util.ExecCommand(ctx, "rbd", args...)
-
+func (rv *rbdVolume) resize(newSize int64) error {
+	image, err := rv.open()
 	if err != nil {
-		return fmt.Errorf("failed to resize rbd image (%w), command output: %s", err, stderr)
+		return err
 	}
+	defer image.Close()
 
-	return nil
+	volSzMiB := util.RoundOffVolSize(newSize)
```

Comment:
newSize is in MiB, but `image.Resize()` takes bytes. The e2e test shows the resize shrunk the volume:
```
Jul 31 03:06:46.241: INFO: At 2020-07-31 02:56:46 +0000 UTC - event for rbd-pvc: {external-resizer rbd.csi.ceph.com } FileSystemResizeRequired: Require file system resize of volume on node
Jul 31 03:06:46.241: INFO: At 2020-07-31 02:56:46 +0000 UTC - event for rbd-pvc: {volume_expand } ExternalExpanding: Ignoring the PVC: didn't find a plugin capable of expanding the volume; waiting for an external controller to process this PVC.
Jul 31 03:06:46.241: INFO: At 2020-07-31 02:56:46 +0000 UTC - event for rbd-pvc: {external-resizer rbd.csi.ceph.com } Resizing: External resizer is resizing volume pvc-e3e15379-56e0-43c3-9cc7-ad2e6e4e330f
Jul 31 03:06:46.241: INFO: At 2020-07-31 02:57:55 +0000 UTC - event for csi-rbd-demo-pod: {kubelet travis-job-8d071cfa-ab4e-4ae4-94a3-be6e13145e8e} VolumeResizeFailed: NodeExpandVolume.NodeExpandVolume failed for volume "pvc-e3e15379-56e0-43c3-9cc7-ad2e6e4e330f" : MountVolume.NodeExpandVolume failed : Expander.NodeExpand failed to expand the volume : rpc error: code = Unknown desc = rbd: resize failed on path /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-e3e15379-56e0-43c3-9cc7-ad2e6e4e330f/globalmount, error: resize of device /dev/rbd0 failed: exit status 1. resize2fs output: resize2fs 1.45.4 (23-Sep-2019)
resize2fs: New size smaller than minimum (12980)
```

Reply:
Strangely, it passed for me during normal resize operation. Will change it.